src/urllib3/fields.py

```diff
@@ -136,9 +178,11 @@ class RequestField:
         An optional filename of the request field. Must be unicode.
     :param headers:
         An optional dict-like object of headers to initially use for the field.
-    :param header_formatter:
-        An optional callable that is used to encode and format the headers. By
-        default, this is :func:`format_header_param_html5`.
+
+    .. versionchanged:: 2.0.0
+        The ``header_formatter`` parameter is deprecated and will
```

Comment:
I don't think it's worth going into that much detail. It would basically say "pass a value that the server you're requesting will accept", which probably adds confusion since servers should be accepting any value.